services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/AuthorizationManagerImpl.java

```diff
@@ -128,6 +111,15 @@ public AuthorizationStatus canAccess(UserInfo userInfo, String objectId, ObjectT
 					return AuthorizationManagerUtil.AUTHORIZED;
 				}
 				return AuthorizationManagerUtil.accessDenied("Only ACT member can perform this action.");
+			case MEMBERSHIP_INVITATION:
+				MembershipInvtnSubmission mis = membershipInvtnSubmissionDAO.get(objectId);

```

Comment:
instead of getting the entire object, add a new query (if not already in place) to get just the invitee ID for the invitation.

Reply:
The logic for authorizing access to invitations is complex enough that it needs to be factored into it's own method (and tested there).  The logic is:
- if the user is the invitee and the access type is READ or DELETE, then they are approved.
- if the user has admin' access (represented as TEAM_MEMBERSHIP_UPDATE access) to the team for which membership is to be granted and the access type is READ or DELETE, then they are approved.
- if the user is a Synapse administrator then access is approved.
- if the user is anonymous then access is denied.